src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -271,9 +269,11 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 
                     foreach (var named in namespaceOrTypeMembers)
                     {
-                        if (emittedTypeName.InferredArity == named.Arity && named.MangleName)
+                        if (emittedTypeName.InferredArity == named.Arity &&
+                            named.MangleName &&
+                            named.MetadataName == emittedTypeName.TypeName)
```

Comment:
>named.MetadataName == emittedTypeName.TypeName [](http://example.com/codeflow?start=28&length=46)

Is there a test backing this change?

Reply:
Yes. This part is tested where I added an assert of ``C`1`` being null. Since ``C`1`` is a mangled name algorithm of its search follows this code path